src/dependencies.jl

```diff
@@ -42,9 +55,31 @@ srcdir(dep) = joinpath(depsdir(dep),"src")
 libdir(provider, dep) = [libdir(dep), libdir(dep)*"32", libdir(dep)*"64"]
 bindir(provider, dep) = bindir(dep)
 
-successful_validate(l,p) = true
+successful_validate(l, p=nothing) = true
+
+function BinDeps.isexecutable(filepath)
+    if isfile(filepath)
+        res = false
+        @unix_only begin
+            res = ccall(:access, Cint, (Ptr{Uint8}, Cint), filepath, 1) == 0
+        end
+        # unfortunately there is no easy way to tell if this process can execute a file on Windows
```

Comment:
The best we could do is ensure the file type ends with one of the PATHEXT extensions, and optionally check files that end in ".exe" for the executable marker (`MZ`). I'll implement that.